packages/core/admin/admin/src/hooks/useSettingsMenu/constants.js

```diff
@@ -21,6 +21,33 @@ export const LINKS_CE = {
       to: '/settings/transfer-tokens?sort=name:ASC',
       id: 'transfer-tokens',
     },
+    ...((!window.strapi.features.isEnabled(
+      window.strapi.features.SSO) && window.strapi.promoteEnterpriseFeatures
+    )
```

Comment:
this is quite hard to read, i've spent a while understanding that what you're doing is:

> Check if the feature is NOT enabled & the `promoteEnterpriseFeatures` is true because if the feature is enabled in EE mode we want the _actual_ route to be part of array. So only checking EE mode could lead to unexpected behaviour

If this is true you need to comment this so it's clear for everyone.